docs/advanced/UsageWithReactRouter.md

```diff
@@ -1,4 +1,158 @@
 # Usage with React Router
 
-Sorry, but we're still writing this doc.  
-Stay tuned, it will appear in a day or two.
+So you want to do routing with your Redux app. You can use it with the react-router library. Redux will be the source of truth for your data and react-router will be the source of truth for your URL. In most of the cases, **it is fine** to have them separate unless if you need to time travel and rewind actions that triggers the change URL.
+
+## Redux-router, react-router-redux, react-router
+Before starting, let's clarify the different routing libraries.
+
+### Redux-router
+Redux-router is an **experimental** library, it lets you keep entirely the state of your url inside your redux store. It has the same API with React Router API but has a smaller community support than react-router.
+
+### React-router-redux
+react-router-redux creates binding between your redux app and react-router and it keeps them in sync. Without this binding, you will not be able to rewind the actions with Time Travel. Unless you need this, React-router and Redux can operates completely apart.
+
+
+## Installing react-router
```

Comment:
Let's always refer to it as React Router in writing, not `react-router` (unless we are talking about npm command).